### DISCUSSION TITLE
minimum opengl version

### DISCUSSION THREAD

**Discussion by bjlittle** [marcomusy/vedo#344] [Ideas]
Hey @marcomusy 

Thanks for such an awesome package :beers: 

Simple question: what is the minimum recommended OpenGL version required by `vedo`?

Many thanks :+1:

**Comment by marcomusy** [comment]
Hi Bill, thanks!
`vedo` builds on top of vtk, so i guess it depends on the vtk version. For version 8.1.2 the minimum required opengl seems to be 1.1 (but might need to recompile vtk using  option VTK_RENDER_BACKEND).

**Comment by bjlittle** [comment]
Thanks Marco for the super fast response :runner::dash: 

If I have spare capacity, I'd love to be able to contribute to `vedo`... is there an easy way in to the codebase for me? 

Do you have some super, super simple issues that I could _try_ to contribute towards and move forwards?

That might be a tangible and beneficial way for me to begin to understand what's going on under the awesome hood of `vedo`

**Comment by bjlittle** [comment]
BTW please do close this issue now. Many thanks! :+1:

**Comment by marcomusy** [comment]
Thanks Bill for your interest! Obviously you're hired! :)
I'm planning to make a new release today if if manage to fix a couple of things.. so probably the best way to start could be to run the already existing examples and check if you can spot anything that seems weird/wrong or simply counter intuitive.
It would be extremely useful to have a "fresh look" from someone who is relatively new to the package!
To upgrade to the dev version:
`pip install -U git+https://github.com/marcomusy/vedo.git`
then
```
cd examples
./run_all.sh
```
feel free to focus only on anything specific you like though! no pressure! :)
Let me know if you need any help,
M.

PS: btw have you used `vedo` for some project already?

**Comment by marcomusy** [comment]
@bjlittle 
I migrated this to discussions - it's probably the right place...

**Comment by bjlittle** [comment]
Good luck with the release Marco :+1: 

Okay, thanks for the guidance, I'll try to make time to dip my toes :wink: 

I realise how valuable it is to have a fresh perspective... regardless of experience (clearly I'm spinning up and looking for an excuse to become more aware of VTK at a deeper level... yup, I'm a noob to VTK, I'm not going to pretend otherwise, but we've all got to start somewhere :wink:)

I'm particularly keen on using a VTK based technology that gives us interoperability between Python, either through the likes of your lovely `vedo` and/or `pyvista`, and `paraview`.

My focus is meteorologically based cubed-sphere quad mesh analytics and visualisation. Matplotlib is super awesome, but sadly just doesn't scale for the high resolution meshes that I (and my team) want to work with i.e., quad meshes containing up to ~6.5-7M nodes. Hence the paradigm shift away from the likes of matplotlib to a technology that can leverage the obvious benefits of OpenGL and GPU acceleration. Hence why I've landed on your doorstep, so to speak.

I started looking around the Scientific Python ecosystem for suitable packages about a year ago and `pyvista` and `vtkplotter` very quickly caught my eye... and I've just been tracking progress since then really, when I can. Now we're at the stage of wanting to build and integrate with a suitable package.

So in summary, that's roughly the context of where I'm coming from, HTH.

Since I've now got your attention... why the rebrand of `vtkplotter` to `vedo`? What was the motivation behind that? Also, how are you using it in a professional capacity? Clearly you're fully vested in `vedo`, given the level of your single handed contributions to the package (which is rather insane BTW :astonished:)... are you curating `vedo` full-time? Is it underpinning other scientific package/s that you're involved with?

**Comment by marcomusy** [comment]
Thanks Bill for the info.
7M vertices sounds  a bit scary also for vtk probably. In `vedo` there's already something like that but would surely need a more clever implementation to handle that many nodes!
```python
from vedo import *
show(TessellatedBox(), Sphere(quads=True), N=2, sharecam=False)
```
![Screenshot from 2021-03-17 16-14-41](https://user-images.githubusercontent.com/32848391/111491211-f64d1880-873b-11eb-87ce-97f12998cf86.png)

> why the rebrand of vtkplotter to vedo? What was the motivation behind that? Also, how are you using it in a professional capacity? 

i liked the name game with the 3d inverted e :) besides I didn't want it to be strictly linked to vtk in the name.  
also pyvista went through to same (i think it was originally called `vtkInterface` if i'm not wrong)

> Clearly you're fully vested in vedo, given the level of your single handed contributions to the package (which is rather insane BTW astonished)... are you curating vedo full-time? 

not at all, `vedo` is just one side activity, my main focus is the embryology of limb development.
I've been developing it because we made , and still make, extensive use of 3d data (volumetric and polygonal surface data) so it all started by trying to put a whole load of sparse scripts into a coherent framework. Then I realized that this could be useful to others for other purposes..

**Comment by bjlittle** [comment]
This is an example of using pyvista to show a C48 cube-sphere Sea Surface Temperature timeseries interactively.

C48 simply means 6 panels x 48 cells x 48 cells = 13,824 node mesh

I'm keen to play with vedo to see what's the experience is like

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/2051656/111588234-2b4c8000-87bb-11eb-8ab7-ef5ad3859209.gif)
